src/mount/extensions/common_deps.clj

```diff
@@ -4,28 +4,64 @@
   (:require [mount.lite :as mount]
             [mount.utils :as utils]))
 
-(defn- transitive
-  "Given a graph and a node, returns all the transitive nodes."
-  [graph node]
-  (loop [unexpanded (graph node)
-         expanded #{}]
-    (if-let [[node & more] (seq unexpanded)]
-      (if (contains? expanded node)
-        (recur more expanded)
-        (recur (concat more (graph node))
-               (conj expanded node)))
-      expanded)))
+;; topsort-component and topsort adapted from loom
+;; https://github.com/aysylu/loom
+
+(defn topsort-component
+  "Topological sort of a component of a (presumably) directed graph.
+  Returns nil if the graph contains any cycles."
+  ([successors start]
+     (topsort-component successors start #{} #{}))
+  ([successors start seen explored]
+     (loop [seen seen
+            explored explored
+            result ()
+            stack [start]]
+       (if (empty? stack)
+         result
+         (let [v (peek stack)
+               seen (conj seen v)
+               us (remove explored (successors v))]
+           (if (seq us)
+             (when-not (some seen us)
+               (recur seen explored result (conj stack (first us))))
+             (recur seen (conj explored v) (conj result v) (pop stack))))))))
+
+(defn topsort
+  "Topological sort of a directed acyclic graph (DAG). Returns nil if
+  g contains any cycles."
+  ([g]
+   (loop [seen #{}
+          result ()
+          [n & ns] (seq (keys g))]
+     (if-not n
+       result
+       (if (seen n)
+         (recur seen result ns)
+         (when-let [cresult (topsort-component
+                             #(get g %) n seen seen)]
+           (recur (into seen cresult) (concat cresult result) ns))))))
+  ([g start]
+   (if start
+     (topsort-component #(get g %) start)
+     (topsort g))))
```

Comment:
This gives a very reliable ordering. I didn't want to pull in the entirety of [Loom](https://github.com/aysylu/loom), which it why I copied the code and adapted it for the existing simple map-based graph.